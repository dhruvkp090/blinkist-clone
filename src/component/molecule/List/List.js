import React from 'react';
import ButtonItem from '../Button/ButtonItem';
import ButtonGroup from '@material-ui/core/ButtonGroup';

const List = (props) => {
    const {listItems, linkTo,   ...others} = props;
    return (
        <div>
            <ButtonGroup orientation={props.orientation}>
                {listItems.map((listItem, index) => (<ButtonItem children={listItem} onClick={props.restore_list} {...others} />))}
            </ButtonGroup>
        </div>
    );
};

export default List;
