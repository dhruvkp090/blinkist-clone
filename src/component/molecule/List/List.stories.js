import React from 'react';
import List from './List';


export default {
    title: "Molecule/List",
    component: List,
};

const Template = args => (<List {...args} /> );


export const NavBarElements = Template.bind({});
NavBarElements.args = {
    listItems: ["Explore", "My Library", "Add Book"],
    color: 'primary',
    href: '#',
    variant: "text",
    orientation: "horizontal"
};

export const FooterList = Template.bind({});
FooterList.args = {
    listItems: ["Book lists", "What is Nonfiction?", "What to Read Next", "Benefits of Reading", "Pricing"],
    color: 'secondary',
    href: '#',
    variant: "text",
    orientation: "vertical"
};