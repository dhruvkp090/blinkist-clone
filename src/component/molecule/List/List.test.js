import React from 'react';
import {render} from '@testing-library/react';
import List from './List';
import '@testing-library/jest-dom';

const listItems= ["Explore", "My Library", "Add Book"];

it('Snapshot horizontal list', () => {
    const { asFragment } = render(<List orientation= "horizontal" listItems={listItems} />);

    expect(asFragment()).toMatchSnapshot();
});

it('Snapshot vertical list', () => {
    const { asFragment } = render(<List orientation= "vertical" listItems={listItems} />);

    expect(asFragment()).toMatchSnapshot();
});