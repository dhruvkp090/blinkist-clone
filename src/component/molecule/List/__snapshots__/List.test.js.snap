// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot horizontal list 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiButtonGroup-root"
      role="group"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
        orientation="horizontal"
        role="button"
        style="justify-content: center;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Explore
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
        orientation="horizontal"
        role="button"
        style="justify-content: center;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          My Library
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
        orientation="horizontal"
        role="button"
        style="justify-content: center;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add Book
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Snapshot vertical list 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiButtonGroup-root MuiButtonGroup-vertical"
      role="group"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlined"
        orientation="vertical"
        role="button"
        style="justify-content: center;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Explore
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlined"
        orientation="vertical"
        role="button"
        style="justify-content: center;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          My Library
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlined"
        orientation="vertical"
        role="button"
        style="justify-content: center;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add Book
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
